examples/multi_physics/piezo_elasticity_micro.py

```diff
@@ -0,0 +1,336 @@
+        'fixed_u': ('Corners', {'u.all': 0.0}),
+        'fixed_r': ('Gamma_ms', {'r.all': 0.0}),
+        'fixed_r1_s1': ('Gamma_s1', {'r.0': 1.0}),
+        'fixed_r0_s1': ('Gamma_s1', {'r.0': 0.0}),
+        'fixed_r1_s2': ('Gamma_s2', {'r.0': 1.0}),
+        'fixed_r0_s2': ('Gamma_s2', {'r.0': 0.0}),
+    }
+
+    integrals = {
+        'i2': 2,
+    }
+
+    solvers = {
+        'ls_d': ('ls.scipy_direct', {}),
+        'ls_i': ('ls.scipy_iterative', {}),
+        'ns_ea6': ('nls.newton', {'eps_a': 1e6, 'eps_r': 1e-3,}),
+        'ns_ea0': ('nls.newton', {'eps_a': 1e0, 'eps_r': 1e-3,}),
+    }
+
+    coefs = {
+        'A1': {
+            'status': 'auxiliary',
+            'requires': ['pis_u', 'corrs_rs'],
+            'expression': 'dw_lin_elastic.i2.Ymc(elastic.D, U1, U2)',
+            'set_variables': [('U1', ('corrs_rs', 'pis_u'), 'u'),
+                              ('U2', ('corrs_rs', 'pis_u'), 'u')],
+            'class': cb.CoefSymSym,
+        },
+        'A2': {
+            'status': 'auxiliary',
+            'requires': ['corrs_rs'],
+            'expression': 'dw_diffusion.i2.Ym(piezo.d, R1, R2)',
+            'set_variables': [('R1', 'corrs_rs', 'r'),
+                              ('R2', 'corrs_rs', 'r')],
+            'class': cb.CoefSymSym,
+        },
+        'A': {
+            'requires': ['c.A1', 'c.A2'],
+            'expression': 'c.A1 + c.A2',
+            'class': cb.CoefEval,
+        },
+        'vol': {
+            'regions': ['Ym', 'Yc1', 'Yc2'],
+            'expression': 'd_volume.i2.%s(svar)',
+            'class': cb.VolumeFractions,
+        },
+        'eps0': {
+            'requires': [],
+            'expression': '%e' % eps0,
+            'class': cb.CoefEval,
+        },
+        'filenames': {},
+    }
+
+    requirements = {
+        'pis_u': {
+            'variables': ['u'],
+            'class': cb.ShapeDimDim,
+        },
+        'pis_r': {
+            'variables': ['r'],
+            'class': cb.ShapeDim,
+        },
+        'corrs_rs': {
+            'requires': ['pis_u'],
+            'ebcs': ['fixed_u', 'fixed_r'],
+            'epbcs': ['p_ux', 'p_uy', 'p_uz', 'p_rx', 'p_ry', 'p_rz'],
+            'equations': {
+                'eq1':
+                    """dw_lin_elastic.i2.Ymc(elastic.D, v, u)
+                     - dw_piezo_coupling.i2.Ym(piezo.g, v, r)
+                   = - dw_lin_elastic.i2.Ymc(elastic.D, v, Pi_u)""",
+                'eq2':
+                    """
+                     - dw_piezo_coupling.i2.Ym(piezo.g, u, s)
+                     - dw_diffusion.i2.Ym(piezo.d, s, r)
+                     = dw_piezo_coupling.i2.Ym(piezo.g, Pi_u, s)""",
+            },
+            'set_variables': [('Pi_u', 'pis_u', 'u')],
+            'class': cb.CorrDimDim,
+            'save_name': 'corrs_rs',
+            'dump_variables': ['u', 'r'],
+            'solvers': {'ls': 'ls_i', 'nls': 'ns_ea6'},
+        },
+    }
+
+    # define requirement and equations for conductors
```

Comment:
requirement?

Reply:
--> # define requirements and coefficients related to conductorsefficients related to conductors